tests/test_gui.py

```diff
@@ -0,0 +1,266 @@
+"""
+test objects
+* main
+* components
+* control
+* monitor
+* plot
+* status_bar
+
+test types
+* user interaction
+* flooding
+
+
+"""
+
+import pytest
+import pdb
+
+from pytestqt import qt_compat
+from pytestqt.qt_compat import qt_api
+
+from vent import gui
+from vent.gui import styles
+from vent.gui import widgets
+from vent.coordinator.coordinator import get_coordinator
+
+from PySide2 import QtCore, QtGui, QtWidgets
+
+import numpy as np
+
+##################################
+# Test user interaction
+# Simulate user actions with the whole intact gui
+
+# turn off gui limiting
+gui.limit_gui(False)
+
+n_samples = 100
+decimals = 5
+global_minmax_range = (0, 100)
+global_safe_range = (25, 75)
+
+@pytest.fixture
+def generic_minmax():
+    """
+    Make a (min, max) range of values that has some safe space in between.
+
+    ie. min can be between 0 and 0.25*(minmax_range[1]-minmax_range[0])
+    and max can similarly be above the 75% mark of the range
+    Returns:
+
+    """
+    def _generic_minmax():
+        abs_min = np.random.rand() * (global_minmax_range[1]-global_minmax_range[0]) * 0.25 + \
+                  global_minmax_range[0]
+        abs_max = np.random.rand() * (global_minmax_range[1]-global_minmax_range[0]) * 0.25 + \
+                  global_minmax_range[0] + (global_minmax_range[1]-global_minmax_range[0]) * 0.75
+        abs_min, abs_max = np.round(abs_min, decimals), np.round(abs_max, decimals)
+        return abs_min, abs_max
+    return _generic_minmax
+
+@pytest.fixture
+def generic_saferange():
+    """
+    Make a (min, max) range of values that has some safe space in between.
+
+    ie. min can be between 0 and 0.25*(minmax_range[1]-minmax_range[0])
+    and max can similarly be above the 75% mark of the range
+    Returns:
+
+    """
+    def _generic_saferange():
+        abs_min = np.random.rand() * (global_safe_range[1]-global_safe_range[0]) * 0.25 + \
+                  global_safe_range[0]
+        abs_max = np.random.rand() * (global_safe_range[1]-global_safe_range[0]) * 0.25 + \
+                  global_safe_range[0] + (global_safe_range[1]-global_safe_range[0]) * 0.75
+        abs_min, abs_max = np.round(abs_min, decimals), np.round(abs_max, decimals)
+        return abs_min, abs_max
+    return _generic_saferange
+
+
+def test_gui_launch(qtbot):
+    assert qt_api.QApplication.instance() is not None
+
+    app = qt_api.QApplication.instance()
+    app.setStyle('Fusion')
+    app.setStyleSheet(styles.DARK_THEME)
+    app = styles.set_dark_palette(app)
+
+    coordinator = get_coordinator(sim_mode=True, single_process=True)
+    vent_gui = gui.Vent_Gui(coordinator)
+    #app, vent_gui = launch_gui(coordinator)
+    qtbot.addWidget(vent_gui)
+    vent_gui.status_bar.start_button.click()
+
+    # wait for a second to let the simulation spin up and start spitting values
+    qtbot.wait(5000)
+
+    assert vent_gui.isVisible()
+
+# def test_gui_launch_mp(qtbot):
```

Comment:
disabled this because remotecoordinator would get a connectionrefused error

Reply:
Reminder: after you merge the latest master, you can enable it.